Content.Client/UserInterface/Controls/ResizableControl.cs

```diff
@@ -0,0 +1,151 @@
+            || (CurrentDrag = GetDragModeFor(args.RelativePosition)) == DragMode.None)
+            return;
+
+        if ((CurrentDrag & AllowedResizeDirection) != CurrentDrag)
+        {
+            CurrentDrag = DragMode.None;
+            return;
+        }
+
+        DragOffsetTopLeft = args.PointerLocation.Position / UIScale - Position;
+        DragOffsetBottomRight = Position + Size - args.PointerLocation.Position / UIScale;
+    }
+
+    protected override void KeyBindUp(GUIBoundKeyEventArgs args)
+    {
+        base.KeyBindUp(args);
+
+        if (args.Function != EngineKeyFunctions.UIClick)
+            return;
+
+        DragOffsetTopLeft = DragOffsetBottomRight = Vector2.Zero;
+        CurrentDrag = DragMode.None;
+
+        UserInterfaceManager.KeyboardFocused?.ReleaseKeyboardFocus();
+    }
+
+    protected override void MouseMove(GUIMouseMoveEventArgs args)
+    {
+        base.MouseMove(args);
+
+        if (!Resizable)
+            return;
+
+        if (CurrentDrag == DragMode.None)
+        {
+            var cursor = CursorShape.Arrow;
+            var previewDragMode = GetDragModeFor(args.RelativePosition);
+            if ((previewDragMode & AllowedResizeDirection) == CurrentDrag)
+                return;
+
+            cursor = previewDragMode switch
+            {
+                DragMode.Top or DragMode.Bottom => CursorShape.VResize,
+                DragMode.Left or DragMode.Right => CursorShape.HResize,
+                (DragMode.Bottom | DragMode.Left) or (DragMode.Top | DragMode.Right) => CursorShape.Crosshair,
+                (DragMode.Bottom | DragMode.Right) or (DragMode.Top | DragMode.Left) => CursorShape.Crosshair,
+                _ => cursor,
+            };
+
+            DefaultCursorShape = cursor;
+        }
+        else
+        {
+            var (left, top) = Position;
+            var (right, bottom) = Position + SetSize;
+
+            if (float.IsNaN(SetSize.X))
+                right = Position.X + Size.X;
+            if (float.IsNaN(SetSize.Y))
+                bottom = Position.Y + Size.Y;
+
+            if ((CurrentDrag & DragMode.Top) == DragMode.Top)
+                top = Math.Min(args.GlobalPosition.Y - DragOffsetTopLeft.Y, Math.Min(bottom, bottom - MinSize.Y));
+            else if ((CurrentDrag & DragMode.Bottom) == DragMode.Bottom)
+                bottom = Math.Max(args.GlobalPosition.Y + DragOffsetBottomRight.Y, Math.Max(top, top + MinSize.Y));
+
+            if ((CurrentDrag & DragMode.Left) == DragMode.Left)
+                left = Math.Min(args.GlobalPosition.X - DragOffsetTopLeft.X, Math.Min(right, right - MinSize.X));
+            else if ((CurrentDrag & DragMode.Right) == DragMode.Right)
+                right = Math.Max(args.GlobalPosition.X + DragOffsetBottomRight.X, Math.Max(left, left + MinSize.X));
+
+            var rect = new UIBox2(left, top, right, bottom);
+            LayoutContainer.SetPosition(this, rect.TopLeft);
+            SetSize = rect.Size;
```

Comment:
Wh-what? Is there... Is there not a vector for mouse movements? Okay I looked, and it's already in the format of a Vector.
So Args.Relative contains the position it is being moved *to*, and args.globalPosition is the position being moved *from*. So I'm not sure that hardcoding all of these direction conditionals is entirely necessary.

For example, to move the container to the new position, I think you just need
```
            LayoutContainer.SetPosition(this, args.GlobalPosition + args.RelativePosition);
```

Reply:
You can try this out and ignore it if I'm wrong about this or if there's another underlying reason why all the directional code is here.